File: frontend/src/api/recetteEngrais.ts
import client from './client'

export interface RecetteEngraisLigne {
  id_ligne?:    number
  id_produit:   number
  dosage:       number
  unite?:       string   // mL/L | g/L
  ordre:        number
  // enrichis depuis le backend
  nom_produit?: string
  type_produit?: string
}

export interface RecetteEngrais {
  id_recette:   number
  nom_recette:  string
  type_recette?: string   // Arrosage | Pulvérisation
  periode?:     string
  semaine?:     number
  ph_cible?:    number
  notes?:       string
  lignes:       RecetteEngraisLigne[]
}

export type RecetteEngraisCreate = Omit<RecetteEngrais, 'id_recette'>
export type RecetteEngraisUpdate = Partial<RecetteEngraisCreate>

export const recetteEngraisAPI = {
  getAll:    ()                                             => client.get<RecetteEngrais[]>('/recettes-engrais/'),
  getById:   (id: number)                                   => client.get<RecetteEngrais>(`/recettes-engrais/${id}`),
  create:    (data: RecetteEngraisCreate)                   => client.post<RecetteEngrais>('/recettes-engrais/', data),
  update:    (id: number, data: RecetteEngraisUpdate)       => client.put<RecetteEngrais>(`/recettes-engrais/${id}`, data),
  delete:    (id: number)                                   => client.delete(`/recettes-engrais/${id}`),
  exportCSV: ()                                             => client.get('/recettes-engrais/export/csv', { responseType: 'blob' }),
  importCSV: (file: File) => {
    const form = new FormData(); form.append('file', file)
    return client.post('/recettes-engrais/import', form, { headers: { 'Content-Type': 'multipart/form-data' } })
  },
}

File: backend/app/schemas/recette_engrais.py
"""Schemas Pydantic pour RecetteEngrais"""
from pydantic import BaseModel, ConfigDict
from typing import Optional, List


class RecetteEngraisLigneBase(BaseModel):
    id_produit:   int
    dosage:       float
    unite:        Optional[str] = None   # mL/L | g/L
    ordre:        int = 0


class RecetteEngraisLigneCreate(RecetteEngraisLigneBase):
    pass


class RecetteEngraisLigneRead(RecetteEngraisLigneBase):
    id_ligne:     int
    nom_produit:  Optional[str] = None   # enrichi depuis ProduitEngrais
    type_produit: Optional[str] = None   # enrichi depuis ProduitEngrais
    model_config  = ConfigDict(from_attributes=True)


# ── Recette principale ────────────────────────────────────────────────────────

class RecetteEngraisBase(BaseModel):
    nom_recette:  str
    type_recette: Optional[str]  = None   # Arrosage | Pulvérisation
    periode:      Optional[str]  = None
    semaine:      Optional[int]  = None   # 1-20
    ph_cible:     Optional[float]= None
    notes:        Optional[str]  = None


class RecetteEngraisCreate(RecetteEngraisBase):
    lignes: List[RecetteEngraisLigneCreate] = []


class RecetteEngraisUpdate(BaseModel):
    nom_recette:  Optional[str]  = None
    type_recette: Optional[str]  = None
    periode:      Optional[str]  = None
    semaine:      Optional[int]  = None
    ph_cible:     Optional[float]= None
    notes:        Optional[str]  = None
    lignes:       Optional[List[RecetteEngraisLigneCreate]] = None


class RecetteEngraisRead(RecetteEngraisBase):
    id_recette: int
    lignes:     List[RecetteEngraisLigneRead] = []
    model_config = ConfigDict(from_attributes=True)
